File: C#1/Homeworks/05-Conditional Statements-Homework/08DigitAsWord/DigitAsWord.cs
using System;
using System.Threading;
using System.Globalization;

//Write a program that read a digit (0-9) from the console, and depending on the input, 
//shows the digit as a word (in English).
//•Print "not a digit" in case of invalid input.
//•Use a switch statement.

//Input
//•The input consists of one line only, which contains the digit.

//Output
//•Output a single line - should the input be a valid digits, print the english word for the digits. 
//Otherwise, print "not a digit".

//Constraints
//•The input will never be an empty line.

class DigitAsWord
 {
     static void Main()
     {
        Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;

        int n;
        string input = Console.ReadLine();
        if (int.TryParse(input, out n))
        {
            switch (n)
            {
                case 0: Console.WriteLine("zero"); break;
                case 1: Console.WriteLine("one"); break;
                case 2: Console.WriteLine("two"); break;
                case 3: Console.WriteLine("three"); break;
                case 4: Console.WriteLine("four"); break;
                case 5: Console.WriteLine("five"); break;
                case 6: Console.WriteLine("six"); break;
                case 7: Console.WriteLine("seven"); break;
                case 8: Console.WriteLine("eight"); break;
                case 9: Console.WriteLine("nine"); break;
                default: Console.WriteLine("not a digit"); break;
            }
        }
        else
        {
            Console.WriteLine("not a digit");
        }
     }
}

File: JS/HW/Conditional-Statements/digit-as-word.js
//Write a script that asks for a digit (0-9), and depending on the input, shows the digit as a word (in English).
// Print not a digit in case of invalid input. Use a switch statement.

function solve(args){
    var digit = +args[0];
    var word;
    
    switch(digit){
        case 0 : word = "zero";break;
        case 1 : word = "one";break;
        case 2 : word = "two";break;
        case 3 : word = "three";break;
        case 4 : word = "four";break;
        case 5 : word = "five";break;
        case 6 : word = "six";break;
        case 7 : word = "seven";break;
        case 8 : word = "eight";break;
        case 9 : word = "nine";break;
        default : word = "not a digit";break;
    }

    console.log(word);

}

solve(['2']);
solve(['1']);
solve(['0']);
solve(['5']);
solve(['-0.1']);
solve(['hi']);
solve(['9']);
solve(['10']);